reader.c

```diff
@@ -127,15 +175,31 @@ void free_nsq_reader(struct NSQReader *rdr)
         LL_FOREACH(rdr->conns, conn) {
             nsqd_connection_disconnect(conn);
         }
+        if (rdr->address != NULL) {
+            free(rdr->address);
+        }
         LL_FOREACH(rdr->lookupd, nsqlookupd_endpoint) {
             free_nsqlookupd_endpoint(nsqlookupd_endpoint);
         }
         free(rdr->topic);
         free(rdr->channel);
+        free(rdr->cfg);
         free(rdr);
     }
 }
 
+int nsq_reader_connect_to_nsqlookupd(struct NSQReader *rdr)
+{
+    // Stop reconnect timer, use lookupd timer instead
+    ev_timer_stop(rdr->loop, &rdr->reconnect_timer);
+
+    ev_timer_init(&rdr->lookupd_poll_timer, nsq_reader_lookupd_poll_cb, 0., rdr->cfg->lookupd_interval);
+    rdr->lookupd_poll_timer.data = rdr;
+    ev_timer_again(rdr->loop, &rdr->lookupd_poll_timer);
```

Comment:
Now that I see the diff, why don't we just move this code into `nsq_reader_add_nsqlookupd_endpoint` (when you add the first `nsqlookupd` endpoint, it starts the timer and stops the `nsqd` timer)?